src/utils/db.ts

```diff
@@ -0,0 +1,54 @@
+import Dexie from 'dexie';
+import { Theme } from '../redux/appSlice';
+
+import { Todo } from '../redux/todosSlice';
+import {createAsyncThunk} from "@reduxjs/toolkit";
+
+interface AppConfigData {
+  id: number;
+  mode: Theme.LIGHT | Theme.DARK | undefined;
+}
+
+export class ToDoDatabase extends Dexie {
+  todos!: Dexie.Table<Todo, string>;
+  appConfig!: Dexie.Table<AppConfigData, number>;
+
+  constructor(databaseName: string) {
+    super(databaseName);
+    this.version(1).stores({
+      todos: '&++id,text,completed,timestamp',
+      appConfig: '++id,mode',
+    });
+  }
+}
+
+export const db = new ToDoDatabase('todoDatabase');
+
+export const saveTodosState = async (state: Todo[]) => {
+  try {
+    await db.todos.clear();
```

Comment:
Why are you clearing the whole list here? I don't think any database works this way

Reply:
This functionality has been implemented in the following way because DB syncs with Store only in one place - when the 'todos' array is changed. So, should I change this functionality and implement synchronization between DB and Store on each action (addTodo, removeTodo, toggleTodo)?